pkg/auxnetdevice/auxNetDeviceProvider.go

```diff
@@ -0,0 +1,181 @@
+				device.Class.ID, vendorName, productName)
+			if isDefaultRoute, _ := utils.HasDefaultRoute(device.Address); !isDefaultRoute {
+				ap.deviceList = append(ap.deviceList, device)
+			}
+		}
+	}
+	return nil
+}
+
+//nolint:gocyclo
+func (ap *auxNetDeviceProvider) GetFilteredDevices(devices []types.HostDevice, rc *types.ResourceConfig) ([]types.HostDevice, error) {
+	filteredDevice := devices
+	nf, ok := rc.SelectorObj.(*types.AuxNetDeviceSelectors)
+	if !ok {
+		return filteredDevice, fmt.Errorf("unable to convert SelectorObj to AuxNetDeviceSelectors")
+	}
+
+	rf := ap.rFactory
+	// filter by vendor list
+	if nf.Vendors != nil && len(nf.Vendors) > 0 {
+		if selector, err := rf.GetSelector("vendors", nf.Vendors); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by device list
+	if nf.Devices != nil && len(nf.Devices) > 0 {
+		if selector, err := rf.GetSelector("devices", nf.Devices); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by driver list
+	if nf.Drivers != nil && len(nf.Drivers) > 0 {
+		if selector, err := rf.GetSelector("drivers", nf.Drivers); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by auxiliary device type list
+	if nf.AuxTypes != nil && len(nf.AuxTypes) > 0 {
+		if selector, err := rf.GetSelector("auxTypes", nf.AuxTypes); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by PfNames list
+	if nf.PfNames != nil && len(nf.PfNames) > 0 {
+		if selector, err := rf.GetSelector("pfNames", nf.PfNames); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by RootDevices list
+	if nf.RootDevices != nil && len(nf.RootDevices) > 0 {
+		if selector, err := rf.GetSelector("rootDevices", nf.RootDevices); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by linkTypes list
+	if nf.LinkTypes != nil && len(nf.LinkTypes) > 0 {
+		if len(nf.LinkTypes) > 1 {
+			glog.Warningf("Link type selector should have a single value.")
+		}
+		if selector, err := rf.GetSelector("linkTypes", nf.LinkTypes); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter for rdma devices
+	if nf.IsRdma {
+		rdmaDevices := make([]types.HostDevice, 0)
+		for _, dev := range filteredDevice {
+			if dev.(types.NetDevice).IsRdma() {
+				rdmaDevices = append(rdmaDevices, dev)
+			}
+		}
+		filteredDevice = rdmaDevices
+	}
+
+	return filteredDevice, nil
+}
+
+// ValidConfig performs validation of AuxNetDeviceSelectors
+func (ap *auxNetDeviceProvider) ValidConfig(rc *types.ResourceConfig) bool {
+	nf, ok := rc.SelectorObj.(*types.AuxNetDeviceSelectors)
+	if !ok {
+		glog.Errorf("unable to convert SelectorObj to AuxNetDeviceSelectors")
+		return false
+	}
+	if len(nf.AuxTypes) == 0 {
+		glog.Errorf("AuxTypes are not specified")
+		return false
+	}
```

Comment:
do we want for now to support only SF Aux Device type ?

do you see today a use-case of an Aux device that is not SF ?

i was thinking for now to just ensure that AuxTypes has a single value of `"sf"` for now if there are no use-cases currently for other aux device types

Reply:
I'm not aware about use-cases of non-SF aux. devices.
We can limit types to `"sf"` only, but for me it seems no problem that plugin will accept other types (assuming that error from getting SF index is ignored). There is just no known users for other types.